src/xSTIR/cSTIR/include/sirf/STIR/stir_data_containers.h

```diff
@@ -528,6 +536,10 @@ namespace sirf {
 			ptr->fill(0.0f);
 			_data.reset(ptr);
 		}
+		PETAcquisitionDataInMemory(std::unique_ptr<stir::ProjData> uptr_pd)
+		{
+			_data = std::move(uptr_pd);
```

Comment:
this makes little sense to me. it wouldn't necessarily be in memory, would it?

Reply:
So far as I understand, `ProjData::get_subset` returns `unique_ptr` to `ProjDataInMemory`, does it not?

Reply:
it does, but this is a general constructor, where you don't know how it will be called. I guess you'd have to do a check if `*uptr_pd` is a `ProjDataInMemory`, if so, `std::move` would be ok, otherwise it'd need to make a copy.

Reply:
@KrisThielemans  Unfortunately, all my attempts to implement your last suggestion failed. My guess is that something is not quite right with `ProjData::get_subset` (see the strange subset sinograms above).